### DISCUSSION TITLE
Cards retrieved from promise.all have rendered but are generic and virtually blank

### DISCUSSION THREAD

**Discussion by nashvegasnate** [nss-evening-cohort-14/evening-client-side#332] [Capstones!] [answered]
Name: Nathan Keith

Issue: Promise.all pulls specific song cards from particular user's song list, however, the cards are just blank except for the url link and buttons. The songCard component usually renders with title, link, description, however, when the cards populate on the singleList view, they appear like this:

<img width="578" alt="Screen Shot 2021-06-22 at 4 42 59 PM" src="https://user-images.githubusercontent.com/76710923/123003517-e9fadc00-d378-11eb-9b48-c62389777e78.png">

I have never seen this before, but have been searching as to how to correct the problem to no avail. Below are the links I have searched:
https://forum.freecodecamp.org/t/reactjs-my-card-component-is-not-rendering-why/388141
https://stackoverflow.com/questions/56693270/i-cant-create-a-new-card-with-react
https://stackoverflow.com/questions/31424561/wait-until-all-promises-complete-even-if-some-rejected

What I have tried: I can't seem to find anything that addresses the issue directly. No idea what to do. Seems so strange.

Link to branch: 
https://github.com/nashvegasnate/SetList/blob/development/src/views/SingleList.js

**Comment by drteresavasquez** [comment]
Can I get a link of the cards?

**Comment by drteresavasquez** [comment]
@nashvegasnate also make sure you're passing the data to the cards. Instead of just naming your proptypes I encourage you use .isRequired on all reused components to ensure you are passing it the values it needs and can debug more easily.

**Comment by TrinityChristiana** [comment]
[You're passing down listSongsInfo as a prop here:
](https://github.com/nashvegasnate/SetList/blob/development/src/views/SingleList.js#L28-L36)
```jsx
 {listSongs?.map((listSongsInfo) => (
        <SongCard
        key={listSongsInfo?.firebaseKey}
        listSongs={listSongsInfo}
        user={user}
        setListSongs={setListSongs}
        // id={id}
        />
      ))}
```

[And the Card is not looking for a prop called `listSongs`](https://github.com/nashvegasnate/SetList/blob/a0f9ea457751d9d21c9e72ac698635a593cb9e8b/src/components/Cards/SongCard.js#L16-L23)
```jsx
...
function SongCard({
  user,
  song,
  setSongs,
  firebaseKey,
  title,
  lists
}) {
...
```

**Comment by nashvegasnate** [comment, marked as answer]
Thank you for the resolution of this ticket, however, I had requested the assistance of @katherinevfry at the beginning of class tonight and she worked her magic!
So, she basically told me the same thing that you all told me in your response, i.e.:
1) That I needed to pass listSongs as a prop in the SongCard 
2) Include the rest parameter of ...listSongsInfo in the SingleList view
3) Include the rest parameter of ...songInfo in the Songs view

Now everything works as intended!
